### PR TITLE
Full migration to the Next.JS app router

### PR DESCRIPTION
**Pull request by TheDarkThief**
I have fully migrated the Club website away from the pages router to the app router, this took about a month and a half.

If it doesn't run, run npm install, I don't know why but it seems to fix packages not loading

These are the main changes made besides moving things around:

- Panda CSS (most of the work)
- Got rid of Chakara UI in favor of custom Zag.JS components
- Restructured the site slightly so we can have pages with no default styling and pages with
- Got rid of Redis because it caching is built-in to Next.JS app router fetch
- W.I.P Loading screen

### CODE REVIEW

**Comment by jaredscarr** [comment]
you have to run npm install because you added new packages. So when someone pulls these new changes they need to be installed locally.

File: app/(mainSite)/events/page.tsx [hunk 1]
```diff
@@ -0,0 +1,57 @@
+import React, { Suspense } from "react";
+
+import Footer from "../../../components/Footer/Footer";
+import EventCards from "../../../components/PageCard/EventCards";
+import { Box, Center, Grid, styled } from "../../../styled-system/jsx";
+import { css } from "../../../styled-system/css";
+
+// TODO: Update docs
```
Comment (jaredscarr) [inline app/(mainSite)/events/page.tsx]: Can this be done in this PR or are you waiting for some additional information from Jonah?

File: app/(mainSite)/gamedev/page.tsx [hunk 2]
```diff
@@ -1,77 +1,73 @@
...
-import Footer from "../components/Footer/Footer";
-import SEO from "../components/SEO/SEO";
+import Footer from "components/Footer/Footer";
 
-import renpyLogo from "../img/RenpyLogo.png";
-import godotLogo from "../img/GodotIcon.svg";
-import greetingBackground from "../img/Homepage-Background.svg";
+import renpyLogo from "/public/img/RenpyLogo.png";
+import godotLogo from "/public/img/GodotIcon.svg";
+// import greetingBackground from "../../public/img/Homepage-Background.svg";
```
Comment (jaredscarr) [inline app/(mainSite)/gamedev/page.tsx]: Remove if no longer used.

File: lib/Event.ts [hunk 3]
```diff
@@ -146,29 +147,35 @@ function dataToCard(data: EventData): EventCard {
  *                is used as a cache key.
  */
 export async function getEventCards(count: number): Promise<EventCard[]> {
+    // // !!OLD CODE USING NEXT JS CACHE SYSTEM!!
```
Comment (jaredscarr) [inline lib/Event.ts]: Any reason to keep this in the code base? Can we just remove it?

File: lib/Event.ts [hunk 4]
```diff
@@ -146,29 +147,35 @@ function dataToCard(data: EventData): EventCard {
...
+            .then((events) => events!.map(dataToCard));
+        }
 
-    // Save the data in the cache.
-    await redisClient.setex(
-        "cache_event_cards:" + count,
-        HOUR,
-        JSON.stringify(fixedData),
-    );
+    // // !!OLD CODE USING NEXT JS CACHE SYSTEM!!
```
Comment (jaredscarr) [inline lib/Event.ts]: Same as above. Why keep?

File: lib/News.ts [hunk 5]
```diff
@@ -41,53 +40,82 @@ giving people information about the club, directing them towards joining the Dis
...
-    }
+
 }
 
 /**
  * Gets the data needed to display several news cards.
  * @param count - is the maximum number of entries to return.
  */
 export async function getNewsCards(count: number): Promise<NewsItem[] | null> {
+    // // OLD CODE SWITCHED TO USING BUILT IN CACHE
```
Comment (jaredscarr) [inline lib/News.ts]: Same remove or tell us why we should keep this here.

File: lib/News.ts [hunk 6]
```diff
@@ -41,53 +40,82 @@ giving people information about the club, directing them towards joining the Dis
...
-        return JSON.parse(cached);
-    }
+    // const cached = await redisClient.get("cache_news_cards:" + count);
+    // if (cached) {
+    //     return JSON.parse(cached);
+    // }
+
     const news = await getNews(count);
+
+    // // OLD CODE SWITCHED TO BUILT IN CACHE
```
Comment (jaredscarr) [inline lib/News.ts]: Same remove or tell us why we should keep this here.

File: lib/Mariadb.ts [hunk 7]
```diff
@@ -1,12 +1,11 @@
-import mariadb from "mariadb";
-import { NewsItem } from "./News";
+// import mariadb from "mariadb";
```
Comment (jaredscarr) [inline lib/Mariadb.ts]: So does this remove the dependency on Mariabdb? Feel like there needs to be a discussion on this. Did the functionality move somewhere? Are we removing the news portion of the website (I'm in favor of that as I don't think it's maintainable in it's current state. especially if we hand it off).

**Review by jaredscarr — CHANGES_REQUESTED** [review]
Amazing work! I have a few questions and clarifications, but this is almost there. Just some details to iron out before the merge.

File: app/(mainSite)/events/page.tsx (hunk 1 above)
Comment (TheDarkThief) [reply app/(mainSite)/events/page.tsx]: I was just trying to get this out ASAP mainly for the impactathon F.A.Q, I'll add the docs now.

File: lib/Event.ts (hunk 3 above)
Comment (TheDarkThief) [reply lib/Event.ts]: kept just in but I'll remove it :D

File: lib/Event.ts (hunk 4 above)
Comment (TheDarkThief) [reply lib/Event.ts]: removed :D

File: README.md [hunk 8]
```diff
@@ -1,33 +1,36 @@
-[![Contributor Covenant](https://img.shields.io/badge/Contributor%20Covenant-2.1-4baaaa.svg)](code_of_conduct.md)
```
Comment (jaredscarr) [inline README.md]: Why did you remove all this?

File: lib/Event.ts [hunk 9]
```diff
@@ -1,4 +1,4 @@
-import { HOUR, redisClient } from "./redis";
+// import { HOUR, redisClient } from "./redis";
```
Comment (jaredscarr) [inline lib/Event.ts]: delete this please.

File: lib/Mariadb.ts (hunk 7 above)
Comment (jaredscarr) [reply lib/Mariadb.ts]: I believe we agreed this needed to be commented back in for this PR and a separate PR will be made later to remove mariadb

File: lib/Mariadb.ts [hunk 10]
```diff
@@ -1,12 +1,11 @@
...
-  database: process.env.DB_NAME,
-  host: process.env.DB_HOST,
-  user: process.env.DB_USER,
-  port: parseInt(process.env.DB_PORT),
-  password: process.env.DB_PASSWORD,
-  connectionLimit: 5,
-  allowPublicKeyRetrieval: true,
-});
\ No newline at end of file
+// export const pool = mariadb.createPool({
```
Comment (jaredscarr) [inline lib/Mariadb.ts]: Same as above.

File: lib/MockRedis.ts [hunk 11]
```diff
@@ -1,16 +1,16 @@
-/**
- * Creates a simple mock data store for development purposes.
- * This is API compatible with redisClient for only things actually used
- * by the website.
- */
-export class MockRedis {
-    store: Record<string, string> = {};
+// /**
```
Comment (jaredscarr) [inline lib/MockRedis.ts]: Remove

**Review by jaredscarr — COMMENTED** [review]
Almost there. The mariadb code needs commenting back in and I'd like a reason for gutting the README.md doc. THere was a lot deleted that is usedful stuff.

File: README.md (hunk 8 above)
Comment (TheDarkThief) [reply README.md]: huh that's odd

Comment (TheDarkThief) [reply README.md]: that should not have happened

**Comment by uellenberg** [comment]
Awesome, thanks for all the hard work! I've made a few changes at https://github.com/nsccs/club_website/commits/app-router/. Please review them and let me know if there's anything I should change. If it all looks good, I'll merge it into your branch.

I did the following:
* Changed to module aliases, because imports like `import ... from "components/..."` are ambiguous and look like you're importing a dependency and not a local file. These imports now look like `import ... from "@/components/..."`.
* I added the header and footer back to some of the pages that were missing it.
* I fixed up some styling issues, mostly related to mobile compatibility.
* I added a background to the mobile menu so that you can click off of it to make it disappear.

Also, what is the source for the following files:
* `Error_404.png`
* `ChangeThwWorld.mp4`
* `welcome_to_404.mp3`

I just don't want to include any sort of copyrighted content.

---

I haven't done a full run through of all the changes yet. I've just been looking at what the page looks like and modifying it as I go. I'll do a proper review shortly.

Thanks again, and hopefully we can get this merged fairly soon!

@TheDarkThief

**Comment by TheDarkThief** [comment]
Yeah that's cool by me, but my only concern is importing libraries like Zag.JS look like: 
`import * as popover from "@zag-js/popover";`
so it might be confusing importing libraries

On the topic of sources
- I drew Error_404.png myself
- A friend recorded welcome_to_404.mp3
- And I heavily edited ChangeThwWorld which is already heavily edited, and it's a easter egg, making it easily fall under fair use. (contains zero of the original audio)

**Comment by uellenberg** [comment]
I agree, it's still a bit confusing, but at least it isn't ambiguous. That sounds good with the sources, and I'll do a quick pass though in a bit. I'll also add the SEO back to the pages.

**Comment by jaredscarr** [comment]
First I know this was large undertaking so it's no surprise that there are some issues to iron out before we merge this:

Major:

- The mobile-menu pop-over links do not function. 
- Forgive me, but in my opinion the modal pop-over is not aesthetically pleasing. I would prefer the drawer or at least something that drops down under the button and not centered on the device.
- The landing page title and buttons is out of alignment for both mobile desktop view (see below mobile images for example). The desktop looks like the bottom margin for the title pushes the buttons too far down or something like that.

original:
current-mobile-buttons

new (and wrong):
new-bad-spacing-buttons

Minor:
- Events date filter was not copied over. The current site lists events from Feb. of this year until now, the new site lists all events back to 2021.

**Comment by uellenberg** [comment]
Some of the things you mentioned are fixed in my branch, namely the styling issues. I'll look over the other issues that you mentioned. I might try bringing the original drawer back - I think the main issue is that there wasn't a ready-built implementation of it, but I think there are still some things that can help to implement it. Once these are fixed, I think we're good to merge it.
